p2phttp.go

```diff
@@ -150,7 +151,7 @@ func (rt *RoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
 		}
 	}()
 
-	resp, err := http.ReadResponse(bufio.NewReader(conn), r)
+	resp, err := http.ReadResponse(bufio.NewReaderSize(conn, network.MessageSizeMax), r)
```

Comment:
@aarshkshah1992 The 4MB here seems like an arbitrary value?. If so, I don't fully see the need to bring it from core/network (it's an indirection that just hides the actual number).

So why 4 MB? I don't expect the average http response to be 4MB in my usecases. Do you in you usecases?

Also would this means that handling 10 concurrent streams is going to need 40MB of memory allocated rather than 4? That would not seem a good thing when running on small devices or VPCs with 512MB of ram.

Reply:
this should be user configurable.

Reply:
@aarshkshah1992 I have reverted the commit. Next time, please do not directly merge and release.

Reply:
@hsanjuan Sorry ! My bad.